manufacturer-toolkit/src/test/java/org/sdo/sct/mt/OnDieTest.java

```diff
@@ -0,0 +1,246 @@
+// Copyright 2020 Intel Corporation
+// SPDX-License-Identifier: Apache 2.0
+
+package org.sdo.sct.mt;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.rules.TemporaryFolder;
+import org.sdo.sct.BouncyCastleSingleton;
+import org.springframework.core.io.ClassPathResource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.security.cert.CertPath;
+import java.security.cert.CertificateFactory;
+import java.util.Base64;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class OnDieTest {
+
+  @Rule
+  public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  private String cacheDirForTesting = "cachedir";
+  private OnDieCache productionOnDieCache = null;
+  private CertificateFactory certFactory = null;
+
+  @Before
+  void initCertFactory() throws Exception {
+    certFactory = CertificateFactory.getInstance("X509", BouncyCastleSingleton.INSTANCE);
+  }
+
+  MStringParser.ParseResult getParsedMstring() throws IOException {
+    final InputStream stream =
+      getClass().getClassLoader().getResourceAsStream("mstring.dat");
+    byte[] mstringBytes = stream.readAllBytes();
+    return new MStringParser().parse(new String(mstringBytes));
+  }
+
+  byte[] getTestArtifact(String artifactName) throws Exception {
+    final InputStream stream =
+      getClass().getClassLoader().getResourceAsStream(artifactName);
+    return stream.readAllBytes();
+  }
+
+  OnDieCache getProductionOnDieCache() throws Exception {
+    if (productionOnDieCache != null) {
+      return productionOnDieCache;
+    }
+    tempFolder.create();
+
+    productionOnDieCache = new OnDieCache(
+      tempFolder.getRoot().getAbsolutePath(),
+      true,
+      "https://tsde.intel.com/content/OnDieCA/certs/,https://tsci.intel.com/content/OnDieCA/crls/");
+
+    return productionOnDieCache;
+  }
+
+
+  @Test
+  @DisplayName("OnDie mstring parse test")
+  void testOnDieMstringParse() throws Exception {
+    MStringParser.ParseResult parseResult =
+      assertDoesNotThrow(() -> getParsedMstring());
+  }
+
+  @Test
+  @DisplayName("OnDie cache download test")
+  void testOnDieCacheDownload() throws Exception {
+
+    tempFolder.create();
+    try {
+      OnDieCache onDieCache = getProductionOnDieCache();
+
+      assertNotNull(onDieCache.getCertOrCrl(
+        "https://tsde.intel.com/content/OnDieCA/=/certs/TGL_00001846_OnDie_CA.crl"));
```

Comment:
It might be a good idea to define the base URL elsewhere, so that it would be easy if we have to change this URL later. It would also minimize unintentional typos.

Reply:
for now I want to keep them within each test since they might be unique values for each test. The URL should never change since this is public facing and could be used by many applications.